### PR TITLE
Release notes v0.36 - dynamical Lie algebras

### PR DESCRIPTION
**Pull request by trbromley**
**Context:**

We have added dynamical Lie algebra support to PennyLane and want to advertise it to the user.

**Description of the Change:**

Adds a section in the release notes.

### CODE REVIEW

File: doc/releases/changelog-0.36.0.md [hunk 1]
```diff
@@ -159,73 +159,80 @@
...
+* The dynamical Lie algebra (DLA) of a set of operators captures the range of unitary evolutions
+  that the operators can generate. In v0.36 of PennyLane, we have added support for calculating
+  important DLA concepts including:
+
+  * A new `qml.lie_closure` function to compute the Lie closure of a list of operators, providing
+    one way to obtain the DLA.
+    [(#5161)](https://github.com/PennyLaneAI/pennylane/pull/5161)
+    [(#5169)](https://github.com/PennyLaneAI/pennylane/pull/5169)
+
+    For a list of operators `ops = [op1, op2, op3, ..]`, one computes all nested commutators between `ops` until no new operators are generated from commutation.
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:171]: Most of the changes are here, the rest is just adding two spaces for indent.

File: doc/releases/changelog-0.36.0.md [hunk 2]
```diff
@@ -159,73 +159,80 @@
...
+    Given a DLA `g`, we can now compute its center. The `center` is the collection of operators that commute with _all_ other operators in the DLA.
+
+    ```pycon
+    >>> g = [X(0), X(1) @ X(0), Y(1), Z(1) @ X(0)]
+    >>> qml.center(g)
+    [X(0)]
+    ```
+
+  To help explain these concepts, check out the
+  [dynamical Lie algebras demo](https://pennylane.ai/qml/demos/tutorial_liealgebra).
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:235]: This is also new

File: doc/releases/changelog-0.36.0.md [hunk 3]
```diff
@@ -159,73 +159,80 @@
...
+
+  * A new `qml.lie_closure` function to compute the Lie closure of a list of operators, providing
+    one way to obtain the DLA.
+    [(#5161)](https://github.com/PennyLaneAI/pennylane/pull/5161)
+    [(#5169)](https://github.com/PennyLaneAI/pennylane/pull/5169)
+
+    For a list of operators `ops = [op1, op2, op3, ..]`, one computes all nested commutators between `ops` until no new operators are generated from commutation.
+    All these operators together form the DLA, see e.g. section IIB of [arXiv:2308.01432](https://arxiv.org/abs/2308.01432).
+
+    Take for example the following ops
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.36.0.md:174]:
```suggestion
    Take for example the following operators:
```

File: doc/releases/changelog-0.36.0.md [hunk 4]
```diff
@@ -159,73 +159,80 @@
...
+  * Computing the structure constants (the adjoint representation) of a dynamical Lie algebra.
+    [(5406)](https://github.com/PennyLaneAI/pennylane/pull/5406)
+
+    For example, we can compute the adjoint representation of the transverse field Ising model DLA.
+
+    ```pycon
+    >>> dla = [X(0) @ X(1), Z(0), Z(1), Y(0) @ X(1), X(0) @ Y(1), Y(0) @ Y(1)]
+    >>> structure_const = qml.structure_constants(dla)
+    >>> structure_constp.shape
+    (6, 6, 6)
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.36.0.md:220]: 😈 😈 😈 😈 😈 😈 😈 😈

Comment (isaacdevlugt) [inline doc/releases/changelog-0.36.0.md:220]: I have no idea if this is correct or not 😆. But, I have no idea what a structure constant is. Is there anything we can explain better here without getting too verbose, or is this okay for users who are familiar with the concept already?

File: doc/releases/changelog-0.36.0.md [hunk 5]
```diff
@@ -159,73 +159,80 @@
...
+    ```
+
+  * Computing the center of a dynamical Lie algebra.
+    [(#5477)](https://github.com/PennyLaneAI/pennylane/pull/5477)
+
+    Given a DLA `g`, we can now compute its center. The `center` is the collection of operators that commute with _all_ other operators in the DLA.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.36.0.md:226]: Small gripe that it should be "centre" 🇨🇦, but will allow "center" 🇺🇸

File: doc/releases/changelog-0.36.0.md (hunk 4 above)
Comment (trbromley) [reply doc/releases/changelog-0.36.0.md:220]: We could recommend checkout out the docs!

File: doc/releases/changelog-0.36.0.md [hunk 6]
```diff
@@ -159,73 +159,80 @@
...
+
+    For example, we can compute the adjoint representation of the transverse field Ising model DLA.
+
+    ```pycon
+    >>> dla = [X(0) @ X(1), Z(0), Z(1), Y(0) @ X(1), X(0) @ Y(1), Y(0) @ Y(1)]
+    >>> structure_const = qml.structure_constants(dla)
+    >>> structure_constp.shape
+    (6, 6, 6)
+    ```
+
```
Comment (trbromley) [inline doc/releases/changelog-0.36.0.md:222]:
```suggestion
    Visit the [documentation of qml.structure_constants](https://docs.pennylane.ai/en/stable/code/api/pennylane.structure_constants.html)
    to understand how structure constants are a useful way to represent a DLA.

```

File: doc/releases/changelog-0.36.0.md (hunk 5 above)
Comment (trbromley) [reply doc/releases/changelog-0.36.0.md:226]: This was a conscious decision to follow more closely the precedent in the literature.

**Review by trbromley — COMMENTED** [review]
Thanks @isaacdevlugt

**Pull request merged**